zcm/julia/ZCM.jl

```diff
@@ -15,320 +35,253 @@ end
-    return ccall(("zcm_stop", "libzcm"), Void, (Ptr{Native.Zcm},), zcm.zcm);
-end
-export stop;
+stop(zcm::Zcm) = ccall(("zcm_stop", "libzcm"), Void, (Ptr{Native.Zcm},), zcm)
 
-function handle(zcm::Zcm)
-    return ccall(("zcm_handle", "libzcm"), Cint, (Ptr{Native.Zcm},), zcm.zcm);
-end
-export handle;
+handle(zcm::Zcm) = ccall(("zcm_handle", "libzcm"), Cint, (Ptr{Native.Zcm},), zcm)
 
 function handle_nonblock(zcm::Zcm)
-    return ccall(("zcm_handle_nonblock", "libzcm"), Cint,
-                 (Ptr{Native.Zcm},), zcm.zcm);
+    ccall(("zcm_handle_nonblock", "libzcm"), Cint,
+          (Ptr{Native.Zcm},), zcm.zcm)
 end
-export handle_nonblock;
 
 # RRR: go over the signedness of all these types from within zcm ... some of them are dumb
 type LogEvent
-    event   ::Ptr{Native.EventLogEvent};
-    num     ::Int64;
-    utime   ::Int64;
-    channel ::String;
-    data    ::Array{UInt8};
+    event   ::Ptr{Native.EventLogEvent}
+    num     ::Int64
+    utime   ::Int64
+    channel ::String
+    data    ::Array{UInt8}
 
     function LogEvent(event::Ptr{Native.EventLogEvent})
-        instance  = new();
-        instance.event = event;
+        instance  = new()
+        instance.event = event
 
         if (event != C_NULL)
-            loadedEvent = unsafe_load(event);
+            loadedEvent = unsafe_load(event)
 
-            instance.num   = loadedEvent.eventnum;
-            instance.utime = loadedEvent.timestamp;
+            instance.num   = loadedEvent.eventnum
+            instance.utime = loadedEvent.timestamp
             if (loadedEvent.channel != C_NULL)
-                instance.channel = unsafe_string(loadedEvent.channel, loadedEvent.chanlen);
+                instance.channel = unsafe_string(loadedEvent.channel, loadedEvent.chanlen)
             else
                 instance.channel = ""
             end
             if (loadedEvent.data != C_NULL)
-                instance.data    = unsafe_wrap(Array, loadedEvent.data, loadedEvent.datalen);
+                instance.data    = unsafe_wrap(Array, loadedEvent.data, loadedEvent.datalen)
             else
-                instance.data    = [];
+                instance.data    = []
             end
         end
 
         # user can force cleanup of their instance by calling `finalize(zcm)`
         finalizer(instance, function(event::LogEvent)
                                 if (event.event != C_NULL)
                                     ccall(("zcm_eventlog_free_event", "libzcm"), Void,
-                                          (Ptr{Native.EventLogEvent},), event.event);
-                                    event.event = C_NULL;
+                                          (Ptr{Native.EventLogEvent},), event.event)
+                                    event.event = C_NULL
                                 end
-                            end);
+                            end)
 
-        return instance;
+        return instance
     end
 end
-export LogEvent;
+export LogEvent
```

Comment:
Should we put this at the top as well?

Reply:
yup, hadn't done the EventLog stuff yet, but I will now. See 6e352f4